server/repository/src/db_diesel/sync_buffer.rs

```diff
@@ -71,12 +72,16 @@ impl<'a> SyncBufferRowRepository<'a> {
 
     #[cfg(feature = "postgres")]
     pub fn upsert_one(&self, row: &SyncBufferRow) -> Result<(), RepositoryError> {
-        diesel::insert_into(sync_buffer_dsl::sync_buffer)
+        let statement = diesel::insert_into(sync_buffer_dsl::sync_buffer)
             .values(row)
             .on_conflict(sync_buffer_dsl::record_id)
             .do_update()
-            .set(row)
-            .execute(&self.connection.connection)?;
+            .set(row);
+
+        // //   Debug diesel query
```

Comment:
Time for this? 
```
if log_enabled!(Level::Debug) {
    debug!("{}", diesel::debug_query::<DBType, _>(&statement).to_string());
}
```

Reply:
I really like this. However the use case for this is log things when tests fail, unfortunately tests don't 'initialise' logger. So I am not sure how to do this best apart from commenting out the println, i've made an issue though: https://github.com/openmsupply/open-msupply/issues/346